### PR TITLE
feat: Show send error message inside the SendModal

### LINKED ISSUE #15828: Displaying sending transaction error in native popup
# Bug Report

## Description

This happened when I wanted to send SNTs, but I had no ETH to pay for gas

image

## Steps to reproduce

Have an account with eg. 2SNTs only. Try to send 1 SNT.

#### Expected behavior

Display error message in Send dialog or a nice popup.

#### Actual behavior

Show ugly native popup.

### Additional Information

- Status desktop version: master
- Operating System: macos

### PR DESCRIPTION
**Pull request by Cuteivist**
Closes #15828

### What does the PR do

Instead of native window error message, show error message inside the SendModal. 
It was based on similar router error visuals.
Additionally parsing was added so message is more clear, without obvious text that it is from status-go and createMultiTx API function.

**NOTE** this is not final product, just simple improvement over current really bad state. 
I'm more than happy to hear suggestions for improvements.

### Affected areas

Send Modal

### Screenshot of functionality (including design for comparison)

image

### CODE REVIEW

File: ui/imports/shared/popups/send/SendModal.qml [hunk 1]
```diff
@@ -65,13 +65,6 @@ StatusDialog {
     property int loginType
     property bool showCustomRoutingMode
 
-    property MessageDialog sendingError: MessageDialog {
```
Comment (caybro) [inline ui/imports/shared/popups/send/SendModal.qml:68]: You can remove the `import QtQuick.Dialogs 1.3` from the top of the file now

File: ui/imports/shared/popups/send/SendModal.qml [hunk 2]
```diff
@@ -153,6 +150,7 @@ StatusDialog {
         readonly property bool isSelectedHoldingValidAsset: !!selectedHolding && selectedHoldingType === Constants.TokenType.ERC20
 
         onSelectedHoldingChanged: {
+            d.sendError = ""
```
Comment (caybro) [inline ui/imports/shared/popups/send/SendModal.qml:153]: Shouldn't you also clear the `sendError` on recipient change? I can't see that anywhere below

Comment (caybro) [reply ui/imports/shared/popups/send/SendModal.qml:153]: Or maybe `onReCalculateSuggestedRoute` is simply enough for everything?

**Review by caybro — COMMENTED** [review]
LGTM overall

File: src/app_service/service/transaction/service.nim [hunk 3]
```diff
@@ -206,6 +206,14 @@ QtObject:
       error "error: ", errDescription
       return
 
+  proc extractRpcErrorMessage(self: Service, errorMessage: string): string =
+    var startIndex = errorMessage.find("message:")
```
Comment (saledjenic) [inline src/app_service/service/transaction/service.nim:210]:
```suggestion
    let startIndex = errorMessage.find("message:")
```

File: src/app_service/service/transaction/service.nim [hunk 4]
```diff
@@ -206,6 +206,14 @@ QtObject:
...
+    if startIndex < 0:
+      return errorMessage
+    startIndex += 8
+    var endIndex = errorMessage.rfind("]")
```
Comment (saledjenic) [inline src/app_service/service/transaction/service.nim:214]:
```suggestion
    const endIndex = errorMessage.rfind("]")
```

**Review by saledjenic — APPROVED** [review]

File: ui/imports/shared/popups/send/SendModal.qml [hunk 5]
```diff
@@ -725,8 +781,8 @@ StatusDialog {
                 if (error.includes(Constants.walletSection.authenticationCanceled)) {
                     return
                 }
-                sendingError.text = error
-                return sendingError.open()
+                d.sendError = error
```
Comment (micieslak) [inline ui/imports/shared/popups/send/SendModal.qml:784]: What about the translations of errors? Shouldn't we take only e.g. type of error as enum from the backend and provide translated strings on the UI side?

Comment (saledjenic) [reply ui/imports/shared/popups/send/SendModal.qml:784]: Good point, look at the line `753` and `754` what we do for router errors, we can do something similar here, but that requires some statusgo work as well. That can be one step of optimizing the sending part on the statusgo side I am planning to work on soon.

Also, just spotted, we have 2 `Connections` with the same target (lines `748` and `775`) we can remove one and just have 2 slots in a single one.

**Review by caybro — APPROVED** [review]

**Comment by dlipicar** [comment]
@Khushboo-dev-cpp we probably want the same thing for Swap, right?

**Comment by Khushboo-dev-cpp** [comment]
> @Khushboo-dev-cpp we probably want the same thing for Swap, right?

afaik I understand we dont show this system error message in swap at all, we rely on the router v2 error messages only. So maybe not needed in case of swap. @Cuteivist can you tell me how I can test this? I'll try it on swap to see how it behaves.

**Pull request merged**